deepdrr/vol.py

```diff
@@ -159,11 +160,139 @@ def from_nifti(
+            world_from_anatomical (Optional[geo.FrameTransform], optional): position the volume in world space. If None, uses identity. Defaults to None.
+            use_cached (bool, optional): [description]. Use a cached segmentation if available. Defaults to True.
+            cache_dir (Optional[Path], optional): Where to load/save the cached segmentation. If None, use the parent dir of `path`. Defaults to None.
+
+        Returns:
+            Volume: an instance of a deepdrr volume
+        """
+        path = Path(path)
+        stem = path.name.split('.')[0]
+
+        if cache_dir is None:
+            cache_dir = path.parent
+
+        # Multi-frame dicoms store all slices of a volume in one file.
+        # they must specify the necessary dicom tags under
+        # https://dicom.innolitics.com/ciods/enhanced-ct-image/enhanced-ct-image-multi-frame-functional-groups
+        assert path.is_file(), 'Currently only multi-frame dicoms are supported. Path must refer to a file.'
+        logger.info(f'loading Dicom volume from {path}')
+
+        # reading the dicom dataset object
+        ds = dcmread(path)
+
+        # extracting all needed tags TODO add try exepts
+        frames = ds.PerFrameFunctionalGroupsSequence
+        shared = ds.SharedFunctionalGroupsSequence[0]
+        num_slices = len(frames)
+        first_slice_position = np.array(frames[0].PlanePositionSequence[0].ImagePositionPatient)
+        last_slice_position = np.array(frames[-1].PlanePositionSequence[0].ImagePositionPatient)
+        RC = np.array(shared.PlaneOrientationSequence[0].ImageOrientationPatient).reshape(2, 3).T
+        PixelSpacing = np.array(ds.SharedFunctionalGroupsSequence[0].PixelMeasuresSequence[0].PixelSpacing)
+        SliceThickness = np.array(ds.SharedFunctionalGroupsSequence[0].PixelMeasuresSequence[0].SliceThickness)
+        offset = ds.SharedFunctionalGroupsSequence[0].PixelValueTransformationSequence[0].RescaleIntercept
+        scale = ds.SharedFunctionalGroupsSequence[0].PixelValueTransformationSequence[0].RescaleSlope
+
+        # make user aware that this is only tested on windows
+        if ds.Manufacturer != "SIEMENS":
+            logger.warning("Multi-frame loading has only been tested on Siemens Enhanced CT DICOMs."
+                           "Please verify everything works as expected.")
+
+        # read the 'raw' data array
+        raw_data = ds.pixel_array.astype(np.float32)
+        hu_values = raw_data * scale + offset
+
+        '''
+        EXPLANATION - indexing conventions
+
+        According to dicom (C.7.6.3.1.4 - Pixel Data) slices are of shape (Rows, Columns) 
+        => must be (j, i) indexed if we define i == horizontal and j == vertical.
+        => we want to conform to the (i, j, k) layout and therefore move the axis of the data array
+        '''
+
+        # convert data to our indexing convention (k, j, i) -> (j, i, k)
+        hu_values = hu_values.transpose((2, 1, 0)).copy()
```

Comment:
@benjamindkilleen Do you agree with this approach? I believe the implementation in `load_dicom.py` does the same.

Reply:
Yes, I believe this is correct to follow the convention in the kernel.